### DISCUSSION TITLE
Getting `nginx_ingress_controller_requests` metric

### DISCUSSION THREAD

**Discussion by JanCizmar** [nginx/kubernetes-ingress#4415] [Q&A] [answered]
Hi,

the kubernetes.io version of nginx ingress controller exposes the `nginx_ingress_controller_requests` metrics, which provides a status code via label.

Can I configure nginxinc controller to provide the same metric? I am getting random 502s and I need to monitor it.

Thanks!

**Comment by haywoodsh** [comment]
Hi @JanCizmar, the NGINX Ingress Controller exposes all the metrics from NGINX/NGINX Plus itself. You can see the list of exported metrics [here](https://github.com/nginxinc/nginx-prometheus-exporter#exported-metrics). Could `nginx_http_requests_total` or `nginxplus_http_requests_total` be what you are looking for?

**Reply by JanCizmar** [reply]
Well, all the metrics labeled by status are part of NGINX plus, which I am currently not using.

**Comment by brianehlert** [comment]
That could be possible using access logging as described here: https://www.nginx.com/blog/using-nginx-logging-for-application-performance-monitoring/

**Reply by JanCizmar** [reply, marked as answer]
Thanks. I actually just used Loki to store the logs from nginx ingress controller and then used this query to get the rate of 502 responses. 
```
sum(rate({container="nginx-ingress"} |~ "\"\\s502\\s"[$__interval]))
```

**Reply by brianehlert** [reply]
quick question - 
do you scrape the access logs from the ingress controller pod logs?  
Or do you use the NGINX native access log syslog streaming to send logs to a Loki collector?
Or, would you have a preference if you could?

**Reply by JanCizmar** [reply]
I am using controller pod logs. It was there out of the box.
